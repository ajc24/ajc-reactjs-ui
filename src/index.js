export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
