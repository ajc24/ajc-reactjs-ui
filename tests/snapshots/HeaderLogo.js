import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { HeaderLogo } from '../../src';
import testImage from '../../stories/images/storybook-header-logo.png';

describe('HeaderLogo', () => {
  describe('Default props and rendering - Image src as an import of image data', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="banner">
          <HeaderLogo id="test-logo-id" imageSrc={testImage} />
        </div>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Default props and rendering - Image src as a string', () => {
    let jsonSnapshot;

    beforeAll(() => {
      const testImagePath = '../../stories/images/storybook-header-logo.png';
      jsonSnapshot = TestDev.createSnapshot(
        <div role="banner">
          <HeaderLogo id="test-logo-id" imageSrc={testImagePath} />
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
