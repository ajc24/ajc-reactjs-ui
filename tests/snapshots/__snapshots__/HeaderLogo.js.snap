// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogo Default props and rendering - Image src as a string verifies the snapshot for the component 1`] = `
<div
  role="banner"
>
  <img
    className="ajc-header-logo"
    id="test-logo-id"
    onLoad={[Function]}
    role="presentation"
    src="../../stories/images/storybook-header-logo.png"
  />
</div>
`;

exports[`HeaderLogo Default props and rendering - Image src as an import of image data verifies the snapshot for the component 1`] = `
<div
  role="banner"
>
  <img
    className="ajc-header-logo"
    id="test-logo-id"
    onLoad={[Function]}
    role="presentation"
    src={Object {}}
  />
</div>
`;
